tenant_schemas_celery/integration_test.py

```diff
@@ -112,3 +112,33 @@ def test_restoring_schema_name(setup_tenant_test):
             update_task.apply_async(args=(dummy2.pk, "updated-name")).get()
 
     assert connection.schema_name == setup_tenant_test["tenant2"].schema_name
+
+
+def test_shared_task_get_schema_name(setup_tenant_test):
+    result = get_schema_name.delay().get(timeout=30)
+    assert result == get_public_schema_name()
+    
+    with tenant_context(setup_tenant_test["tenant1"]):
+        result = get_schema_name.delay().get(timeout=30)
+
+    assert connection.schema_name == setup_tenant_test["tenant1"].schema_name
```

Comment:
You need to use `result` instead of `connection.schema_name`, as the changes from this app should not leak outside of task's submission. That is - your connection's schema should not be different before and after sending a task.

Reply:
thanks! copy/paste error